clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```
@@ -748,13 +747,19 @@ private void maybeThrottle(AbstractResponse response, short apiVersion, String n
      *
      * @param responses The list of responses to update
      * @param now The current time
+     * @throws IOException 
      */
     private void handleCompletedReceives(List<ClientResponse> responses, long now) {
         for (NetworkReceive receive : this.selector.completedReceives()) {
             String source = receive.source();
             InFlightRequest req = inFlightRequests.completeNext(source);
             Struct responseStruct = parseStructMaybeUpdateThrottleTimeMetrics(receive.payload(), req.header,
                 throttleTimeSensor, now);
+            try {
+                receive.close();
+            } catch (IOException e) {
+                log.info("Failed closing NetworkReceive from {}", source);
```

Is it intentional to swallow this exception? At a minimum, we should probably log the exception at DEBUG level.

---

Yes we should log the exception